src/Engines/nativeEngine.ts

```diff
@@ -1506,34 +1507,38 @@ export class NativeEngine extends Engine {
-                return this._native.SAMPLER_MINPOINT_MAGPOINT_MIPLINEAR;
+            case Constants.TEXTURE_NEAREST_NEAREST:
+                return this._native.TEXTURE_NEAREST_NEAREST;
+            case Constants.TEXTURE_LINEAR_LINEAR:
+                return this._native.TEXTURE_LINEAR_LINEAR;
+            case Constants.TEXTURE_LINEAR_LINEAR_MIPLINEAR:
+                return this._native.TEXTURE_LINEAR_LINEAR_MIPLINEAR;
             case Constants.TEXTURE_NEAREST_NEAREST_MIPNEAREST:
-                return this._native.SAMPLER_NEAREST;
+                return this._native.TEXTURE_NEAREST_NEAREST_MIPNEAREST;
             case Constants.TEXTURE_NEAREST_LINEAR_MIPNEAREST:
-                return this._native.SAMPLER_MINLINEAR_MAGPOINT_MIPPOINT;
+                return this._native.TEXTURE_NEAREST_LINEAR_MIPNEAREST;
             case Constants.TEXTURE_NEAREST_LINEAR_MIPLINEAR:
-                return this._native.SAMPLER_MINBILINEAR_MAGPOINT;
+                return this._native.TEXTURE_NEAREST_LINEAR_MIPLINEAR;
             case Constants.TEXTURE_NEAREST_LINEAR:
-                return this._native.SAMPLER_MINBILINEAR_MAGPOINT;
-            case Constants.TEXTURE_NEAREST_NEAREST:
-                return this._native.SAMPLER_NEAREST;
+                return this._native.TEXTURE_NEAREST_LINEAR;
+            case Constants.TEXTURE_NEAREST_NEAREST_MIPLINEAR:
+                return this._native.TEXTURE_NEAREST_NEAREST_MIPLINEAR;
             case Constants.TEXTURE_LINEAR_NEAREST_MIPNEAREST:
-                return this._native.SAMPLER_MINPOINT_MAGLINEAR_MIPPOINT;
+                return this._native.TEXTURE_LINEAR_NEAREST_MIPNEAREST;
             case Constants.TEXTURE_LINEAR_NEAREST_MIPLINEAR:
-                return this._native.SAMPLER_MINPOINT_MAGLINEAR_MIPLINEAR;
-            case Constants.TEXTURE_LINEAR_LINEAR:
-                return this._native.SAMPLER_TRILINEAR;
-            case Constants.TEXTURE_LINEAR_NEAREST:
-                return this._native.SAMPLER_MINPOINT_MAGLINEAR_MIPLINEAR;
-            case Constants.TEXTURE_NEAREST_NEAREST_MIPLINEAR:
-                return this._native.SAMPLER_MINPOINT_MAGPOINT_MIPLINEAR;
+                return this._native.TEXTURE_LINEAR_NEAREST_MIPLINEAR;
             case Constants.TEXTURE_LINEAR_LINEAR_MIPNEAREST:
-                return this._native.SAMPLER_TRILINEAR;
+                return this._native.TEXTURE_LINEAR_LINEAR_MIPNEAREST;
+            case Constants.TEXTURE_LINEAR_NEAREST:
+                return this._native.TEXTURE_LINEAR_NEAREST;
+
+            // Legacy constants
+            case Constants.TEXTURE_NEAREST_SAMPLINGMODE:
+                return this._native.TEXTURE_NEAREST_NEAREST;
+            case Constants.TEXTURE_BILINEAR_SAMPLINGMODE:
+                return this._native.TEXTURE_LINEAR_LINEAR;
+            case Constants.TEXTURE_TRILINEAR_SAMPLINGMODE:
+                return this._native.TEXTURE_LINEAR_LINEAR_MIPLINEAR;

```

Comment:
Is this necessary? They are the same values as the other constants?

Reply:
Ha! No, good catch. I keep forgetting they're just numbers and thinking of them like enums.